File: Tut2_ROS_Publisher_Subscriber/src/first_pub_sub/scripts/SubNode.py
#!/usr/bin/env python3

import rospy
from std_msgs.msg import String

def callback_TopicNode_data(msg):
    rospy.loginfo("Message Received")
    infoMsg ="Sub node is receiving data from " + msg.data
    rospy.loginfo(infoMsg)

if __name__ == '__main__':
    #select Ros node name
    nodeName = "SubNode_rospy"

    #select Ros topic name
    TopicName = "/TopicNodeData"

    #Init Node with its name
    rospy.init_node(nodeName)

    #Configure node node type(Subcriber) ,the topic Name, and callback
    sub = rospy.Subscriber(TopicName,String,callback_TopicNode_data)

    #keep node running
    rospy.spin()

File: Tut2_ROS_Publisher_Subscriber/src/first_pub_sub/src/subNode.cpp
#include <ros/ros.h>
#include <string>
#include <std_msgs/String.h>

void callback_TopicNode_data(const std_msgs::String& msg)
{
    ROS_INFO("Message Received");
    const std::string msg_info("Sub node is receiving data from " + msg.data);
    ROS_INFO("%s",msg_info.c_str());
}

int main(int argc, char **argv)
{
    //select Ros node name
    const std::string nodeName("SubNode_roscpp");

    //select Ros topic name
    const std::string TopicName("/TopicNodeData");

    //Init Node with its name
    ros::init(argc, argv, nodeName);

    //create node handle
    ros::NodeHandle nh;

    //Configure node node type(Subcriber), the topic Name, and queue size
    ros::Subscriber sub(nh.subscribe(TopicName, 1000, callback_TopicNode_data));

    //keep node running
    ros::spin();
}